### ISSUE TITLE
when using ansi-term in spacemacs, some bugs happens no matter in GUI or Terminal 

### ISSUE THREAD

**Issue by freestyle4568** [syl20bnr/spacemacs#2775]
1.When I type some command in the bottom of ansi-term, it always jump up and down.
Just like you type one time, it jump one. I DON not know how to fix it.
2.when I open the vim in ansi-term or use ansi-term in terminal, the color become very ugly.
![spacemacs-bug1](https://cloud.githubusercontent.com/assets/10106571/9518907/303d5e86-4ced-11e5-8c48-a035161d77b0.png)
![spacemacs-bug2](https://cloud.githubusercontent.com/assets/10106571/9518953/84e02a04-4ced-11e5-8a85-9e814cf11556.png)

**Comment by syl20bnr** [comment]
Weird, it is like you are using smart-eshell in ansi-term. Does the behavior you see matches this description ? 

> When you hit enter, the cursor stays on the previous prompt, you can then re-execute the last command verbatim or modify it to correct an error. Only when you edit the last command, the cursor goes to the last prompt.

You can read more about this here: https://github.com/syl20bnr/spacemacs/tree/develop/contrib/shell#enable-em-smart-in-eshell

**Comment by cmccloud** [comment]
For the color scheme problem, try this:
M-x customize-face "term"
untick default bg and default fg colors - all that should be left is inherit

**Comment by freestyle4568** [comment]
sorry man, it does not work still.I has given up....

**Comment by cmccloud** [comment]
It may not be the same issue, but I was having color and formatting problems using ansi-term. There were two issues that I had to solve - the first was solved by running the following command, which I found here: http://rawsyntax.com/blog/learn-emacs-zsh-and-multi-term/

```
tic -o ~/.terminfo /Applications/Emacs.app/Contents/Resources/etc/e/eterm-color.ti
```
Note that I'm on OSX so the location of your eterm-color.ti file will most likely be different, to find it you can run:
```
locate eterm-color.ti
```
If that still doesn't work, try adding this to your emacs config:
```
(setq system-uses-terminfo nil)
```
The second issue was related to custom-set-faces overriding the value of term's background and foreground color face (the same thing I mentioned earlier) - if you want to try again this is the command for it, but it's the same as using the customize menu. You should make sure that's the only line in your custom-set-faces related to term.
```
(custom-set-faces
 '(term ((t (:inherit default)))))
```

Hopefully that fixes your issues, but if not look for me @cmccloud on the gitter chat and I'll see if I can't help.

**Comment by mclearc** [comment]
I had the same problem and the two solutions offered by @cmccloud solved it.

**Comment by freestyle4568** [comment]
thank you very much. But I still solve them. Sorry, I decise to give up spacemacs and turn to eclipse and vim. Maybe emacs is not fitable for me. But thank you

**Comment by freestyle4568** [comment]
I still wish spacemacs will develop more and more convience. Yes, it will become the best emacs eventually, I believe. Thank you for spacemacs, thank you for everybody.I love emacs and I am sure I will come back when its version is 1.0.0.

**Comment by jgoold** [comment]
The comments in this issue address the color / formatting problem described.  Is there a solution to "When I type some command in the bottom of ansi-term, it always jump up and down.
Just like you type one time, it jump one."

I have the same jumping-lines problem (emacs 24.3, RHEL7).  The problem doesn't seem to manifest itself when I run a vanilla emacs configuration (no spacemacs).

**Comment by StreakyCobra** [comment]
@jgoold Can you try if this bug still arise with the last spacemacs release? Maybe also with the `develop` branch if you can :relaxed: ?

**Comment by jgoold** [comment]
This does appear to be fixed in spacemacs-0.104.4.  However, I also moved
to emacs 24.5.1 from emacs 24.3... so I'm not sure which change actually
resolved the issue.

**Comment by StreakyCobra** [comment]
@jgoold Thx for your response :-)

A solution for colors has been given, and the jumping problem should have been solved in the last release. As no one said to have this problem anymore, it  I'm closing it. Feel free to open a new issue in case of troubles, and link this one in it if you think it's related :relaxed:
